Gabriel - admin pages

gdumouli opened the pull request:
Created new fragments for the Admin sub-pages (Java and .xml) and connected the backend to it

gdumouli committed:
Classes of admin's fragments created
Base for connecting admin's fragments created in AdminActivity
New backButton added on Admin's headbar.

gdumouli committed:
Created xml pages for Admin Fragments

gdumouli committed:
Transition between admin fragments work
Also added header text in all fragments

gdumouli committed:
Admin's events page works
New ConfirmDeleteFragment created (to delete document from admin pages)

gdumouli committed:
Admin "View Profiles" work

gdumouli committed:
Displaying all database images in Admin works
'Photo' class was heavily modified for Admin's images

gdumouli committed:
Photo deletion works

ColeDewis commented on .vs/ProjectSettings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
we should update the gitignore: .vs sounds like vscode settings, which shouldn't be uploaded to git

ColeDewis commented on .vs/noRAM/v16/.suo:
same with this file

ColeDewis commented on .vs/slnx.sqlite:
same with this file

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/AdminEventsFragment.java:
```diff
@@ -0,0 +1,92 @@
+package com.example.noram;
```
would be nice to add a top of the file comment as we have for other files

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/AdminImagesFragment.java:
```diff
@@ -0,0 +1,129 @@
+package com.example.noram;
```
top file comment

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/AdminProfilesFragment.java:
```diff
@@ -0,0 +1,101 @@
+package com.example.noram;
```
top comment

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/ConfirmDeleteFragment.java:
```diff
@@ -0,0 +1,72 @@
+package com.example.noram;
```
top comment

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/ConfirmDeleteFragment.java:
```diff
@@ -0,0 +1,72 @@
+     */
+    private void deleteDoc(){
+        docToDelete.delete()
+                .addOnFailureListener(new OnFailureListener() {
```
NIT: tab should be 4 space ideally

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/ConfirmDeleteFragment.java:
```diff
@@ -0,0 +1,72 @@
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.w("AdminDeleteDoc", "Could not delete the document from admin " +
```
same as above

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/ConfirmDeleteFragment.java:
```diff
@@ -0,0 +1,72 @@
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+        return builder
+                .setTitle("Confirm delete?")
```
same as above

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/controller/AttendeeArrayAdapter.java:
```diff
@@ -0,0 +1,69 @@
+package com.example.noram.controller;
```
top comment

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/controller/PhotoArrayAdapter.java:
```diff
@@ -0,0 +1,76 @@
+package com.example.noram.controller;
```
top comment

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/model/Photo.java:
```diff
@@ -62,4 +80,98 @@ public String getPhotoPath() {
+        this.photoProfile = isProfile;
+    }
+
+    public boolean getPhotoProfile(){
```
docstring

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/model/Photo.java:
```diff
@@ -14,6 +28,10 @@
 public class Photo {
```
want to make sure I understand the purpose of these changes here; you've added observer pattern to the photo class so ImageViews can listen be connected to a photo so that when it is updated they are updated.

This doesn't seem bad to me, but what specifically is the use for this? If an admin deletes a profile photo, we'd need to communicate that through the database to other devices, so this wouldn't exactly be able to update that. Is this intended for that part of the app, or just for the admin section?

gdumouli replied:
I made updateWithBitmap() because of the list (grid) of ImageViews in the admin's image section. 
When a view gets created in the list, it needs to be immediately displayed. However, queries to the database are asynchronous. Thus, while waiting for images to get fetched, a placeholder image is displayed until the database's image is loaded.

All of the changes I made were to implement the photos in the Admin section: if the changes are too heavy or unrelated however, I can also make an AdminPhoto subclass and move the implementation.

ColeDewis replied:
Makes sense, I don't have any issue with whatever implementation you found to be best. I do think it might be good to rename the class/make it a bit clearer though - just using "Photo" as the class for this to me suggests that it would be used for photos across the app, when in reality this is specifically for the admin photos. It might be a good idea to rename the class (I don't think anything else is using the Photo class anyways?)

ColeDewis reviewed (changes requested):
Some missing documentation, generally looks good. Would appreciate any clarification on the changes in the Photo class

gdumouli committed:
PR Review

gdumouli committed:
Merge remote-tracking branch 'origin/master' into gabriel-admin-pages

gdumouli committed:
Additional issues for AdminImagesFragment

gdumouli committed:
Further PR Review
Photo renamed to AdminPhoto
Top comments added
.vs files removed (and added in gitignore)

gdumouli committed:
Unit Test for AdminActivity

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/model/Event.java:
```diff
@@ -174,15 +174,15 @@ public ArrayList<Integer> getMilestones() {
 
     /**
-     * Returns Photo of promotional QR code
+     * Returns AdminPhoto of promotional QR code
```
these don't return a photo, they're just QRCode objects.

ColeDewis reviewed (approved):
Thanks for the changes, LGTM, made one small comment on documentation.

gdumouli committed:
PR Review + Admin Page Test

gdumouli committed:
Ignore database's "Cupcake" images in Admin-Images

ColeDewis commented on code/noRAM/app/src/main/java/com/example/noram/AdminImagesFragment.java:
```diff
@@ -100,7 +100,10 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
                         for(StorageReference item : listResult.getItems()){
-                            addPhotoToGrid(item, true);
+                            // add photo if it's not a setting photo
+                            if(!item.getPath().contains("Cupcake")){
```
I think we also want to include lowercase cupcake to catch one of the other photos in cloud storage right now. Also not sure if we want to exclude default profile pictures here, that would need to also exclude "-default"

gdumouli committed:
Improved filtering in Admin-Images page

gdumouli committed:
Merge remote-tracking branch 'origin/master' into gabriel-admin-pages
